test/node/connection-emit-actions.js

```diff
@@ -0,0 +1,221 @@
+      const connection = new jstp.Connection(transport, null, client);
+
+      connection.on('handshake', (error, ok) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(ok, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+      });
+
+      connection.handshake(app.name, null, null, (error) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(connection.username, null, 'username must be null');
+        test.equal(connection.sessionId, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+        connection.close();
+      });
+    });
+  }
+);
+
+test.test('must emit server and client events upon login/password handshake',
+  (test) => {
+    test.plan(7);
+
+    const client = {
+      application: new jstp.Application('jstp', {}),
+    };
+
+    let serverConnection = null;
+    server.once('connect', (serverConn) => {
+      serverConnection = serverConn;
+    });
+
+    const port = server.address().port;
+    const socket = net.connect(port);
+    socket.on('error', () => {
+      test.fail('must create socket and connect to server');
+    });
+    socket.on('connect', () => {
+      serverConnection.on('handshakeRequest',
+        (applicationName, authStrategy) => {
+          test.equal(applicationName, app.name,
+            'application name must match');
+          test.equal(authStrategy, 'login',
+            'auth strategy must be \'login\' with login/password');
+        }
+      );
+
+      const transport = new jstp.net.Transport(socket);
+      const connection = new jstp.Connection(transport, null, client);
+
+      connection.on('handshake', (error, ok) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(ok, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+      });
+
+      connection.handshake(app.name, app.login, app.password, (error) => {
+        test.assertNot(error, 'handshake must not return an error');
+        test.equal(connection.username, app.login, 'username must match');
+        test.equal(connection.sessionId, app.sessionId,
+          'session id must be equal to the one provided by authCallback');
+        connection.close();
+      });
+    });
+  }
+);
+
+test.test('must emit event on call with no arguments and no return value',
+  (test) => {
+    test.plan(5);
+
+    const iface = 'calculator';
+    const methodName = 'doNothing';
+    const args = [];
+    addCallEventCheck(test, server.getClients()[0], iface, methodName, args);
+    connection.on('callback', (error, ok) => {
+      test.assertNot(error, 'callMethod must not return an error');
+      test.strictSame(ok, [], 'Ok contents must match');
+    });
+    connection.callMethod(iface, methodName, args);
+  }
+);
+
+test.test('must emit event on call with no arguments and return value',
```

Comment:
"with no arguments and return value" is a bit misleading: it is not clear if "no" does also relate to "return value".